src/ast/schemaPredicates.test.ts

```diff
@@ -29,4 +29,13 @@ describe('SchemaPredicates', () => {
     expect(schemaPredicates.isFieldNullable('Todo', 'complete')).toBeTruthy();
     expect(schemaPredicates.isFieldNullable('Todo', 'author')).toBeTruthy();
   });
+
+  it('should handle unions of objects', () => {
+    expect(
+      schemaPredicates.isInterfaceOfType('LatestTodoResult', 'Todo')
+    ).toBeTruthy();
+    expect(
+      schemaPredicates.isInterfaceOfType('LatestTodoResult', 'NoTodosError')
```

Comment:
I think we're missing a test for this:

```js
expect(
  schemaPredicates.isInterfaceOfType('Todo', 'NoTodosError')
).toBeFalsy();
```

Also is `LatestTodoResult` compared with `NoTodosError` / `Todo` legal? As in are you able to write this in GraphQL? Because I'm not sure

```graphql
fragment _ on LatestTodoResult {
}
```

Where `LatestTodoResult` is obviously a union, hence a fragment shouldn't be able to match at all 🤔

Reply:
I think that should be ok, but should only yield __typename?

Reply:
Yep that's valid (sorry commented on wrong thread):

![image](https://user-images.githubusercontent.com/296796/68943892-bb844a80-07ac-11ea-8170-1328d708d951.png)